src/ray/raylet/node_manager.cc

```diff
@@ -2855,6 +2855,7 @@ std::optional<syncer::RaySyncMessage> NodeManager::CreateSyncMessage(
 
   rpc::ResourcesData resources_data;
   resources_data.set_should_global_gc(true);
+  resources_data.set_cluster_full_of_actors_detected(resource_deadlock_warned_ >= 1);
   syncer::RaySyncMessage msg;
   msg.set_version(absl::GetCurrentTimeNanos());
   msg.set_node_id(self_node_id_.Binary());
```

Comment:
I feel the code doesn't match the PR description? Which line of code fixes the warning issue?

Reply:
it matches. The old code works like: https://sourcegraph.com/github.com/ray-project/ray@master/-/blob/src/ray/raylet/node_manager.cc?L1816

btw, it's a bad design. we only want it warn once.

this parameter controls this.

Reply:
also here fyi, https://sourcegraph.com/github.com/ray-project/ray@master/-/blob/src/ray/raylet/node_manager.cc?L943-945

Reply:
What about other changes in this PR?